### DISCUSSION TITLE
Has outputDiffMaxSize been deprecated?

### DISCUSSION THREAD

**Discussion by brianng-gh** [vitest-dev/vitest#3534] [Q&A] [answered]
Hello `vitest` team!

When running `vitest` without code coverage report, I get the error below:
```
Could not display diff. It's possible objects are too large to compare.
Try increasing --outputDiffMaxSize option.
```

And when returning back to `vitest` docs, there's no trace of `--outputDiffMaxSize` option anymore.

Is there a new way to allow or increase "large diff" in `vitest` output? Or is there a better or new way? Lastly, I really need to output  a large object to `toEqual` to ensure the returned `prisma` query (with relation) is correct.

**Comment by sheremet-va** [comment]
Yes, diffing was changed in one of the versions. Your Vitest version is low.

**Reply by brianng-gh** [reply]
Thanks @sheremet-va ! FYI, my installed `vitest` version is `vitest 0.29.8` (after running `pnpm list`). I must go higher than `0.29.8`? Also, a bit wary to go to latest, as I am not ready with `vitest` custom dependencies set up, as they may have to upgrade up in sync.

**Reply by sheremet-va** [reply, marked as answer]
If you are on 0.29.8 you can use `outputDiffMaxSize` in the config. If you need different diffing behavior, upgrade to latest.

**Reply by brianng-gh** [reply]
@sheremet-va Interesting, I did try updating `vitest.config.ts` like below (found this from a random online thread), and it was still the same error earlier. 

> If this is incorrect usage, then most likely culprit and I just need to know if it is `true` / `false`, or `10000`, and instead of `Infinity` and so on.

```
export default defineConfig({
  test: {
    ...
    outputDiffLines: Infinity,
    outputDiffMaxLines: Infinity,
    outputDiffMaxSize: Infinity,
    outputTruncateLength: Infinity,
    coverage: {
      ...
    },
  },
});
```

And yes, will plan to upgrade `vitest` to latest, thanks again! 🍻 This might solved the issue despite updated `vitest.config.ts` file.

**Comment by anli-xsigns** [comment]
I'm using vitest 0.34.1 and the following code works:
```
import { defineConfig } from 'vitest/config'
import tsconfigPaths from 'vite-tsconfig-paths'

export default defineConfig({
    outputDiffLines: Infinity,
    outputDiffMaxLines: Infinity,
    outputDiffMaxSize: Infinity,
    outputTruncateLength: Infinity,
    plugins: [tsconfigPaths()]
})
```

However these lines create typescript warnings as the config parameter type doesn't include these options:
```
    outputDiffLines: Infinity,
    outputDiffMaxLines: Infinity,
    outputDiffMaxSize: Infinity,
    outputTruncateLength: Infinity,
```
Can this be fixed?

**Reply by sheremet-va** [reply]
These options are not supported

**Reply by maximilianfixl** [reply]
If I run a test with this configuration in 0.34.6, a shortened text of the difference with ellipsis still appears on the console.

`vitest.config.ts` test options:
```javascript
import { defineConfig } from 'vitest/config'

// This brings in TypeScript, like anli-xsigns already explained: Object literal may only specify
// known properties, and  outputDiffLines  does not exist in
// type  InlineConfig
export default defineConfig({
    test: {
        outputDiffLines: Infinity,
        outputDiffMaxLines: Infinity,
        outputDiffMaxSize: Infinity,
        outputTruncateLength: Infinity,
    }
})

// The same TypeScript error with this configuration
export default defineConfig({
    outputDiffLines: Infinity,
    outputDiffMaxLines: Infinity,
    outputDiffMaxSize: Infinity,
    outputTruncateLength: Infinity,
})
```

configuration test: 

```javascript
test('output should show infinite length of diff', () => {
    const actual = faker.string.alphanumeric(7000)
    const expected = faker.string.alphanumeric(7000)
    expect(actual).toBe(expected)
})
```

Result:

```shell
Expected :MLL0ubddBSx3nTtUmo5vYpFoWxtRvQBuheErHCFC9jaHcOkwLweEjHRauSfXgrmqPyxBZWtd3UmfItwKJy9FjQc5rv2h7UIoxHY8xtHHtJh565CSKRXOI0Yi4R2IEPfAYhzuA2du1NaYSp3WWqrD3W9Vk1FGauu9C12uD1ui26OtZ12LqxPoXyW8uavZizsPXcCpDI096KYrsNBEidfLlSVBza1y5HllEy4SiBRRYUsUD6P3KwW5ueMAQu ...

Actual   :OBWjj0yFzbF1NfR7EZv742xtDTlPm8BRYZd1WMRr6yqTIh2RsiZRdZ8ZblPE8hy6Yb620vX5xHv2XjzdV5qjPzZ3eOUUUUK6mW12eT7avXJKUCUaeEGzIWXkmL4yAjXYe04wVArs4i4ARe0A9iENbmKTpByp2zzlJrs6zagYtUW6Jfk6ekYDJxXTSOd5XMfEV5qazNb82uu07b8bctYr5Zk7Wbmfyf6sRrZruwt1htRiV4UC15rpcEjgEY ...
```

As we see, the strings end at 251 characters. At this point a configuration option for the length of the output of the difference would be useful.

I work with a JetBrains IDE and have the possibility to click on `<Click to see difference>`, which works fine, but ends quickly in a tab chaos during intensive testing.

**Reply by sheremet-va** [reply]
This line is configurable by chaiConfig: https://vitest.dev/config/#chaiconfig-truncatethreshold

**Reply by maximilianfixl** [reply]
Amazing, this works for me! 🍻

```javascript
export default defineConfig({
    test: {
        chaiConfig: {
            truncateThreshold: Infinity,
        },
    },
    plugins: [tsconfigPaths()],
})
```
